src/gui/sharedialog.cpp

```diff
@@ -145,8 +145,10 @@ void ShareDialog::addLinkShareWidget(const QSharedPointer<LinkShare> &linkShare)
 
     connect(linkShare.data(), &Share::serverError, _linkWidgetList.at(index), &ShareLinkWidget::slotServerError);
     connect(linkShare.data(), &Share::shareDeleted, _linkWidgetList.at(index), &ShareLinkWidget::slotDeleteShareFetched);
-    connect(_manager, &ShareManager::linkShareRequiresPassword, _linkWidgetList.at(index), &ShareLinkWidget::slotCreateShareRequiresPassword);
-    connect(_manager, &ShareManager::serverError, _linkWidgetList.at(index), &ShareLinkWidget::slotServerError);
+    if(_manager != nullptr) {
```

Comment:
Missing space before the opening parenthesis, I'd also go for `if (_manager) {` since pointers have a natural and standard truth value.

Reply:
Yeah, that's my style too; Guess this happens it you try to adapt to existing code too much, as has been in the file.